ttnn/cpp/ttnn/operations/data_movement/common/common.cpp

```diff
@@ -55,6 +55,84 @@ ttnn::Tensor pad_to_tile_vol(
     return tensor;
 }
 uint32_t wrap_index(int index, int size) { return index < 0 ? size + index : index; }
+
+ttnn::MemoryConfig create_sharded_memory_config(
+    const ttnn::Shape& shape,
+    const tt::tt_metal::CoreRangeSet& core_grid,
+    const ShardStrategy& strategy,
+    const tt::tt_metal::ShardOrientation& orientation,
+    bool halo,
+    bool use_height_and_width_as_shard_shape,
+    const tt::tt_metal::Layout& layout) {
+    auto is_tile_layout = layout == tt::tt_metal::Layout::TILE;
+
+    auto rank = shape.rank();
+    TT_FATAL(rank >= 2, "rank of tensor to shard must be at least 2.");
+
+    auto tensor_memory_layout = ttnn::TensorMemoryLayout::BLOCK_SHARDED;
+    if (strategy == ShardStrategy::WIDTH) {
+        tensor_memory_layout = ttnn::TensorMemoryLayout::WIDTH_SHARDED;
+    } else if (strategy == ShardStrategy::HEIGHT) {
+        tensor_memory_layout = ttnn::TensorMemoryLayout::HEIGHT_SHARDED;
+    }
+
+    auto shard_orientation = orientation;
+    auto shard_grid = core_grid;
+
+    auto height = shape[-2];
+    auto width = shape[-1];
+    std::array<uint32_t, 2> shard_shape;
+
+    if (use_height_and_width_as_shard_shape) {
+        if (shard_orientation == tt::tt_metal::ShardOrientation::ROW_MAJOR) {
+            shard_shape = {height, width};
+        } else if (shard_orientation == tt::tt_metal::ShardOrientation::COL_MAJOR) {
+            shard_shape = {width, height};
```

Comment:
If `use_height_and_width_as_shard_shape` is set, the code does something completely different, ignoring `core_grid`, `halo`, `layout`, and other arguments. To me this is a clear sign that it should be a separate function

Reply:
Resolving this in favor of one discussion thread.